### DISCUSSION TITLE
Windows: Using older RealTek sound drivers and ASIO4ALL 2.14 for JackTrip

### DISCUSSION THREAD

**Discussion by SynthiaPayne** [jacktrip/jacktrip#1168] [General]
Hello All!

This issue began around 2022, after RealTek updated their Windows sound drivers that separate the headphones and speaker outputs. Before that, the headphone and speaker outputs were combined.

Note: this issue does not pertain to RealTek ASIO, which has been blacklisted. It is the RealTek sound driver for Windows and ASIO4ALL, which is a “wrapper” for the RealTek WDM drivers that make them behave like ASIO.

Issue: Headphone output is not engaged when using the current RealTek drivers and ASIO4ALL.

My workarounds:

Only use an older RealTek driver version 6.0.9254.1 dated 10/19/2021.
Only use ASIO4ALL version 2.14 (version 2.15 is non-functional for JackTrip.)

Details:

Current RealTek drivers for Windows Sound have the speaker and headphone (if connected) configured as two separate outputs. When using ASIO4ALL 2.13, JackTrip does not engage the headphone output and only plays through the speakers.

A previous RealTek driver noted in my workarounds above, combines the speaker and headphone outputs. ASIO4ALL v.2.14 and JackTrip successfully engage the headphones when connected and speaker outputs as needed.

With the current RealTek drivers that separate the headphone and speaker output and ASIO4ALL 2.14:

a. JackTrip does not engage with the headphone and only plays through the speaker, even when the headphone is plugged in and selected.

b. In the VS settings, when tapping “Test Tone”, there is signal but no sound in the only the speaker. There is no change when selecting headphones in the Windows sound settings.

c. In VS settings, when changing to output 3&4, and tapping “Test Tone”, there is sound in the headphones. However, when in the studio interface, there is still no sound in headphones.

d. The settings in ASIO4ALL accommodate 2 channels for the speakers as per the RealTek driver. There is another output labeled, “2nd Output”. This could be the headphone output, however, JackTrip’s output protocol for headphones is not finding it as a second output, but rather possibly as channels 3&4.

Resources for the older working RealTek driver:

Uncompress DIY: The older working RealTek compressed driver files:
[Microsoft Update Catalog](https://www.catalog.update.microsoft.com/Search.aspx?q=6.0.9254.1)

How to install .cab files above:
[How to install CAB file for updates and drivers on Windows 10 - Pureinfotech](https://pureinfotech.com/install-cab-file-windows-10/)

**Comment by mikedickey** [comment]
@SynthiaPayne just curious if this behaves any differently now with 2.2? I changed some things that may be relevant.

**Reply by SynthiaPayne** [reply]
Just seeing this now, and tested with 2.2. Sorry no change. 
Still need to select chs. 3/4 for output to headphones. Ch. 1/2 plays out of speakers only.

Thank you.
